src/Compilers/Shared/BuildClient.cs

```diff
@@ -167,6 +167,24 @@ public Task<RunCompilationResult> RunCompilationAsync(IEnumerable<string> origin
         {
             BuildResponse buildResponse;
 
+            IDisposable npcs = null;
```

Comment:
This block of code is going to run on every platform but it's only necessary on Mono. I think we should extract this out to a separate function, `AreNamedPipesSupported` which essentially executes this when on Mono but returns `true` in all other cases. The helper `PlatformInformation.IsRunningOnMono` can be used for the Mono check.